internal/cli/kraft/run/run.go

```diff
@@ -244,6 +219,35 @@ func (opts *RunOptions) discoverMachineController(ctx context.Context) error {
 	return nil
 }
 
+func (opts *RunOptions) discoverNetworkController(ctx context.Context) error {
+	var err error
+
+	if opts.Network == "" && opts.IP != "" {
+		return fmt.Errorf("cannot assign IP address without providing --network")
```

Comment:
This essentially prevents users from instances without a network. Instead, simply `return nil`.